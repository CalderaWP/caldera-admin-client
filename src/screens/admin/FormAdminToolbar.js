import React from 'react';
import {Component} from '@wordpress/element';
import {createSlotFill, Button} from '@wordpress/components'
import {Admin} from '@caldera-labs/components'
import {Status} from "../../components/Layout/Status";
import PropTypes from 'prop-types'
import types from '../../types'
const {Fill, Slot} = createSlotFill('FormAdminToolbar');

class FormAdminToolbar extends Component {

	static NavBar = ({label, onActive, onDeactive, isActive = false}) => (
		<Fill
			name="FormAdminToolbar.NavBar"
			label={label}
			onActive={onActive}
			onDeactive={onDeactive}
		>
			<li>
				<Button
					isPrimary
					onClick={() => {
						if (isActive) {
							onDeactive();
						} else {
							onActive()
						}

					}}
				>
					{label}
				</Button>
			</li>
		</Fill>
	);




	constructor(props) {
		super(props);
		this.state = {currentItem: null};
		this.handleClick = this.handleClick.bind(this);
		this.proStatusMessage = this.proStatusMessage.bind(this);
	}

	handleClick({props}) {
		if (this.state.currentItem) {
			this.state.currentItem.onDeactive();
		}

		props.onActive();
		this.setState({currentItem: props});
	}

	proStatusMessage(){
		return this.props.isProConnected ? 'Connected' : 'Not Connected';
	}

	render() {
		const {mainStatus} = this.props;
		return (
			<div>
				<Admin.CalderaHeader>
					<Slot
						name="FormAdminToolbar.NavBar"
						fillChildProps={{onClick: this.handleClick}}
					/>
					<li>
						<Status
							className={FormAdminToolbar.classNames.mainStatus}
							message={mainStatus.message}
							success={mainStatus.success}
							show={mainStatus.show}
							updating={mainStatus.updating}
						/>
					</li>
					<li>
						<Status
							className={FormAdminToolbar.classNames.proStatus}
							message={this.proStatusMessage()}
							success={this.props.isProConnected}
							show={this.showProStatus}
						/>
					</li>
				</Admin.CalderaHeader>


			</div>
		);
	}
}

export default FormAdminToolbar;


FormAdminToolbar.propTypes = {
	isProConnected: PropTypes.bool,
	showProStatus: PropTypes.bool,
	mainStatus: types.statusType,
};

FormAdminToolbar.defaultProps = {
	isProConnected: false,
	showProStatus: true,
};

FormAdminToolbar.classNames = {
	proStatus: 'cf-pro-status',
	mainStatus : 'cf-status'
}
