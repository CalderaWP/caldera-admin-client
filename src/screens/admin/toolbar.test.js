import renderer from 'react-test-renderer';
import React from 'react';
import FormAdminToolbar from './FormAdminToolbar'


describe( 'FormAdminToolbar', () => {
	it( 'snapshots', () => {
		const mainStatus = {
			message: '',
				show: false,
				success: false,
				updating: false
		}
		expect(
			renderer.create(
				<FormAdminToolbar
					mainStatus={mainStatus}
				/>
			).toJSON()
		).toMatchSnapshot();
	});


});

