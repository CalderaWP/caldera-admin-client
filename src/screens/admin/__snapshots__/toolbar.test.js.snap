// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAdminToolbar snapshots 1`] = `
<div>
  <div
    className="caldera-editor-header"
  >
    <ul
      className="caldera-editor-header-nav"
    >
      <li
        className="caldera-editor-logo"
      >
        <span
          className="caldera-forms-name"
        >
          Caldera Forms: 
          Caldera Forms
        </span>
      </li>
      <div
        role="presentation"
      />
      <li />
      <li />
    </ul>
  </div>
</div>
`;
